x-pack/plugins/reporting/server/lib/esqueue/worker.js

```diff
@@ -316,6 +324,11 @@ export class Worker extends events.EventEmitter {
 
         return this._claimJob(job)
           .then((claimResult) => {
+            // could not update the job, use the previous object
+            if (claimResult === true) {
```

Comment:
I'm trying to think what the right course of action is here. It got a conflict, which indicates to me that some other process updated the job prior to this one. That probably means that another Kibana / es_queue instance is running the job, right? I'm not sure we actually want to proceed here.

If our *tests* are modifying the job out from under us, we should fix the tests, because I think in the real world, we want to treat conflicts here as if a competing Kibana process is running the report. @Stacey-Gammon maybe you can shed some light on this. I'm not really familiar enough with es_queue to know if I'm right / wrong here.

Reply:
I agree that failing the job in case of conflict is probably the right thing to do. My understanding is that a conflict means another Kibana process has claimed the job and will be responsible for that report generation.

Ideally, `_claimJob` would always return a valid job object, and if it can't, it should throw an error. We'd log the error (not a debug log) and trust that the job will run on another instance, or that the user will see some messaging that something went wrong and re-try. I'm not familiar enough to know if those are safe assumptions, though. It would really help to add a unit test around `_processPendingJobs` to hit all the scenarios.

Reply:
> Ideally, _claimJob would always return a valid job object, and if it can't, it should throw an error. We'd log the error (not a debug log) and trust that the job will run on another instance, or that the user will see some messaging that something went wrong and re-try.

My thoughts as well, and this would make a natural flow when we add types.  

>  It would really help to add a unit test around _processPendingJobs to hit all the scenarios.

Is this something feasible to add to this PR? Then we could maybe confidently change the return value?

Reply:
Yes, as long as the reviewers are on board, I will make that change of return types in this PR and add the unit test

Reply:
Yep. Onboard. The whole true / false / object return type is a bit hard to follow, so this would be a nice improvement.